rom/dev/src/flow/cold_reset/idev_id.rs

```diff
@@ -75,22 +81,35 @@ impl InitDevIdLayer {
         Self::derive_cdi(env, KEY_ID_UDS, KEY_ID_ROM_FMC_CDI)?;
 
         // Derive DICE Key Pair from CDI
-        let key_pair = Self::derive_key_pair(env, KEY_ID_ROM_FMC_CDI, KEY_ID_IDEVID_PRIV_KEY)?;
-
-        // Generate the Subject Serial Number and Subject Key Identifier.
+        let (ecc_key_pair, mldsa_pub_key) = Self::derive_key_pair(
+            env,
+            KEY_ID_ROM_FMC_CDI,
+            KEY_ID_IDEVID_ECDSA_PRIV_KEY,
+            KEY_ID_IDEVID_MLDSA_KEYPAIR_SEED,
+        )?;
+
+        // Generate the Subject Serial Number and Subject Key Identifier for ECC.
         // This information will be used by next DICE Layer while generating
         // certificates
-        let subj_sn = X509::subj_sn(env, &key_pair.pub_key)?;
+        let ecc_subj_sn = X509::subj_sn(env, &ecc_key_pair.pub_key)?;
         report_boot_status(IDevIdSubjIdSnGenerationComplete.into());
 
-        let subj_key_id = X509::idev_subj_key_id(env, &key_pair.pub_key)?;
+        let ecc_subj_key_id = X509::idev_subj_key_id(env, &ecc_key_pair.pub_key)?;
         report_boot_status(IDevIdSubjKeyIdGenerationComplete.into());
 
+        // [TODO] Generate the Subject Serial Number and Subject Key Identifier for MLDSA.
+
         // Generate the output for next layer
         let output = DiceOutput {
-            subj_key_pair: key_pair,
-            subj_sn,
-            subj_key_id,
+            ecc_subj_key_pair: ecc_key_pair,
+            ecc_subj_sn,
+            ecc_subj_key_id,
+            mldsa_subj_key_id: [0; 20],
+            mldsa_subj_key_pair: MlDsaKeyPair {
+                key_pair_seed: KEY_ID_IDEVID_MLDSA_KEYPAIR_SEED,
```

Comment:
Nit: it's a bit inconsistent that `Self::derive_key_pair()` returns a tuple where the ECC key is a complete key pair, but the ML-DSA key is only the public portion, requiring us to make sure that the key ID passed here is the same key ID passed to `Self::derive_key_pair()` above. Instead of that, we could just have `Self::derive_key_pair()` returned a tuple of type `(Ecc384KeyPair, MlDsaKeyPair)`.

(And similarly, `mldsa_key_gen()` could be updated to return a full `MlDsaKeyPair` instead of just the public portion.)

Reply:
Udpated.